File: 0101-symmetric-tree/0101-symmetric-tree.py
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        def chk(left, right):
            if left is None and right is None:
                return True
            if left is None or right is None:
                return False

            return (
                left.val == right.val and
                chk(left.left, right.right) and
                chk(left.right, right.left)
            )


        return chk(root, root)

File: 0101-symmetric-tree/0101-symmetric-tree.cpp
class Solution {
private:
    bool chk(TreeNode* left, TreeNode* right) {
        if (left == NULL && right == NULL) return true;
        if (left == NULL || right == NULL) return false;

        return left->val == right->val && chk(left->left, right->right) && chk(left->right, right->left);
    }

public:
    bool isSymmetric(TreeNode* root) {
        return chk(root, root);
    }
};